app/src/main/java/org/thoughtcrime/securesms/preferences/BlockedContactsActivity.kt

```diff
@@ -18,11 +26,31 @@ class BlockedContactsActivity: PassphraseRequiredActionBarActivity() {
 
     val adapter: BlockedContactsAdapter by lazy { BlockedContactsAdapter(viewModel) }
 
-    fun unblock() {
+    private fun unblock() {
         showSessionDialog {
             title(viewModel.getTitle(this@BlockedContactsActivity))
-            text(viewModel.getMessage(this@BlockedContactsActivity))
-            button(R.string.continue_2) { viewModel.unblock() }
+
+            val contactsToUnblock = viewModel.state.selectedItems
```

Comment:
Ios there a reason the logic was moved out of the VM and into the view? Preferably the view should only be given what it needs to display and all the logic should be contained in the VM.

Reply:
Moved logic to determine what text to use when unblocking contacts into a method in the viewmodel